src/librbd/LibrbdWriteback.cc

```diff
@@ -163,6 +168,8 @@ namespace librbd {
 				 object_no, off, objectx, object_overlap,
 				 bl, snapc, snap_id,
 				 req_comp);
+    if (m_ictx->closing)
+      req->set_op_flags(LIBRADOS_OP_FLAG_FADVISE_DONTNEED);
```

Comment:
@dillaman . For rbd, if there are much client, it's dangerous to use DONTNEED. It's depend on the use mode.
Can we add flag in rbd_close/rbd_flush to indicate the mode?